blockstore/splitstore/debug.go

```diff
@@ -0,0 +1,360 @@
+package splitstore
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime/debug"
+	"strings"
+	"sync"
+	"time"
+
+	"go.uber.org/multierr"
+	"golang.org/x/xerrors"
+
+	blocks "github.com/ipfs/go-block-format"
+	cid "github.com/ipfs/go-cid"
+)
+
+type debugLog struct {
```

Comment:
This is a beautiful piece of code ;-)

Reply:
Maybe a bit less beautiful?

```go
type debugLogOp struct {
    path string

    mu     sync.Mutex
    count  int64
}

type debugLog struct {
    read, write, delete, stack debugLogOp
    stackMap                   map[string]string
}
```

(and then DRY up the constructor as well)

Reply:
Would also let you delete a bunch of these functions. You could even have a "forAll" helper.

Reply:
sure we can do that. I didn't put that much thought in code cosmetics here, it was purely pragmatic while debugging the misses.

Reply:
refactored to DRY in https://github.com/filecoin-project/lotus/pull/6474/commits/f5ae10e3d1b8447f814c7c44082a390cbc4a094d